tutorials/transparent-proxy-and-filtering-on-k8s-with-initializers/index.md

```diff
@@ -0,0 +1,214 @@
+
+Just like in the previous tutorial, the purpose of the [tproxy-sidecar](https://github.com/danisla/kubernetes-tproxy/tree/master/sidecar) container is to create firewall rules in the pod network to block egress traffic. The [tproxy-podwatch](https://github.com/danisla/kubernetes-tproxy/tree/master/cmd/tproxy-podwatch) controller watches for pod changes containing the annotation and automatically add/removes the local firewall `REDIRECT` rules to apply the transparent proxy to the pod.
+
+![architecture diagram](./tproxy_initializers_diagram.png)
+
+**Figure 1.** transparent proxy with initializers architecture diagram
+
+## Objectives
+
+- Create a Kubernetes cluster with initializer and RBAC support using Google Container Engine
+- Deploy the tproxy, tproxy-initializer and the tproxy-podwatch pods using Helm
+- Deploy example apps with annotations to test external access to a Google Cloud Storage bucket
+
+## Before you begin
+
+This tutorial assumes you already have a Google Cloud Platform (GCP) account and are familiar with the high level concepts of Kubernetes Pods and Deployments.
+
+## Costs
+
+This tutorial uses billable components of GCP, including:
+
+- [Google Container Engine](https://cloud.google.com/container-engine/pricing)
+
+Use the [Pricing Calculator](https://cloud.google.com/products/calculator/#id=f52c2651-4b02-4da3-b8cd-fdbca6ad89a9) to estimate the costs for your environment. The resources used in this tutorial are approximately $2.39 per day.
+
+## Checkout the source repository
+
+1. Open [Cloud Shell](https://console.cloud.google.com/cloudshell)
+
+2. Clone the repository containing the code for this tutorial:
+
+        git clone https://github.com/danisla/kubernetes-tproxy
+        cd kubernetes-tproxy
+
+    The remainder of this tutorial will be run from the root of the cloned repository directory.
+
+## Create Container Engine cluster and install Helm
+
+1. Create Container Engine cluster with alpha features enabled, RBAC support, and a cluster version of at least 1.7 to support initializers:
+
+        gcloud container clusters create tproxy-example \
+        --zone us-central1-f \
+        --machine-type n1-standard-1 \
+        --num-nodes 3 \
+        --enable-kubernetes-alpha \
+        --cluster-version 1.7.6 \
+        --no-enable-legacy-authorization
+
+    This command also automatically configures the kubectl command to use the cluster.
+
+2. Create a service account and cluster role binding for Helm to enable RBAC support:
+
+        kubectl create serviceaccount tiller --namespace kube-system
+
+        kubectl create clusterrolebinding tiller-cluster-rule \
+        --clusterrole=cluster-admin \
+        --serviceaccount=kube-system:tiller
+
+3. Install the Helm tool locally in your Cloud Shell instance:
+
+        curl -sL https://storage.googleapis.com/kubernetes-helm/helm-v2.5.1-linux-amd64.tar.gz | tar -xvf - && sudo mv linux-amd64/helm /usr/local/bin/ && rm -Rf linux-amd64
```

Comment:
can you refer to helm docs here? you're hardcoding multiple things that can easily stop working tomorrow.

Reply:
Helm 2.6 has an issue with the initializer object creation. I want to hard code it. The tutorial (any my other tutorials) are under CI to catch breaking changes.